src/common/ChannelChatters.cpp

```diff
@@ -27,8 +31,18 @@ void ChannelChatters::addRecentChatter(const QString &user)
 
 void ChannelChatters::addJoinedUser(const QString &user)
 {
-    auto joinedUsers = this->joinedUsers_.access();
-    joinedUsers->append(user);
+    assertInGuiThread();
+
+    if (getApp()
+            ->getAccounts()
+            ->twitch.getCurrent()
+            ->blockedUserLogins()
+            .contains(user))
+    {
+        return;
```

Comment:
imo moderator status should override & show join/part, even if user was blocked

Reply:
Done in https://github.com/Chatterino/chatterino2/pull/6181/commits/f805385655844209bc2516a1f2e6d125774b60db thanks!